configure.ac

```diff
@@ -62,8 +62,12 @@ PKG_CHECK_MODULES([CRYPTO], [libcrypto >= 1.1.0])
 PKG_CHECK_MODULES([CURL], [libcurl])
 
 # pretty print of devicepath if efivar library is present
-PKG_CHECK_MODULES([EFIVAR], [efivar],,[true])
-AC_CHECK_HEADERS([efivar/efivar.h])
+AC_ARG_WITH([efivar], AS_HELP_STRING([--without-efivar], [Build without efivar library (default: test)]))
+
+AS_IF([test "x$with_efivar" != "xno"], [
+	PKG_CHECK_MODULES([EFIVAR], [efivar])
+	AC_CHECK_HEADERS([efivar/efivar.h])
+])
```

Comment:
shouldn't this be:
```m4
AS_IF([test "x$with_efivar" != "xtest"], [
	PKG_CHECK_MODULES([EFIVAR], [efivar],,[true])
	AC_CHECK_HEADERS([efivar/efivar.h])
],
[test "x$with_efivar" != "xyes"], [
	PKG_CHECK_MODULES([EFIVAR], [efivar],,[true])
	AC_CHECK_HEADERS([efivar/efivar.h])
])
```

Probably don't need the AC_CHECK_HEADERS, that can be dropped.

Reply:
The AC_CHECK_HEADERS can probably be dropped, but the point of the patch is to NOT autoprobe for efivar and include it if found, which the above does. See https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Automagic_dependencies.The way he original patch is setup the default is efivar is expected unless --without-efivar is stated. If efivar is not present, an error is generated, and the builder must either include it or specify --without-efivar to exclude it.

Reply:
Given my approach, which is autoprobe by default, but can be user overridden, Why can't Gentoo's build system add the configure flag `--without-efivar` when building. Then the build system is non-automagic for their build.

Reply:
The current build actually does include efivar, and with your changes above, if --with-efivar is passed, but the efivar library is not present, it just ignores and builds out, rather than error out. If efivar support is requested (either by default or explicitly) then ./configure process should error out if efivar isn't present .

Reply:
> The current build actually does include efivar, and with your changes above, if --with-efivar is passed, but the efivar library is not present, it just ignores and builds out, rather than error out. If efivar support is requested (either by default or explicitly) then ./configure process should error out if efivar isn't present .

That was the behavior I wanted, but that's a trivial fix. Ultimately the behavior should be:

| config options | efi present | result              |
|------------------- |----------------|--------------------|
| auto (default) | yes            | efi in build       |
| auto (default) | no             | efi not in build | 
| --with-efi        | yes            | efi in build       |
| --with-efi       | no              | fail configure   | 
| --without-efi  | yes            | efi not in build |
| --without-efi  | yes            | efi not in build |

Reply:
Yes, that works